### ISSUE TITLE
Add `services:` top-level YAML configuration item which will enable specified services during boot.

### ISSUE THREAD

**Issue by svpernova09** [laravel/homestead#1420]
Primary Use case:

Ship non-default services in an "off" state to save some performance in the VM.

Easily allow users to specify which services should be enabled. Homestead will process as `systemctl <service> enable`

### Default services enabled
* `mailhog`
* `nginx` (unless site type `apache` exists)
* `php*-fpm` version specific FPM
* `postfix` (Relay to `mailhog`)
* `supervisor`

### Services installed but disabled / not running:

* `apache2` (unless site type `apache` exists)
* `beanstalkd` (launches w/ `/etc/init.d/beanstalkd start`)
* `postgresql@9.6-main`
* `postgresql@10-main`
* `postgresql@11-main`
* `postgresql@12-main`
* `redis-server`

**Comment by svpernova09** [comment]
## Enable or Disable Services

Homestead starts several services by default however if your configuration overrides one of these defaults you can specify which services you would like to enable or disable during provisioning. For example if you only needed PostgreSQL 12 your `services:` configuration in `Homestead.yaml` might look like:

    services:
        - enabled:
            - "postgresql@12-main"
        - disabled:
            - "mysql"

This configuration would ensure `postgresql@12-main` service would be enabled and `mysql` would be disabled from starting at boot. The specified services will also be started or stopped based on their location in `enabled` and `disabled` sections.

**Comment by svpernova09** [comment]
Resolved via 414000dcb77dd14b5801867fb5aab6894be6bbe6

**Comment by yazeed** [comment]
> postgresql@12-main

Shouldn't it be as follows, without dashes for the service list?
```
services:
    - enabled:
        "postgresql@12-main"
    - disabled:
        "mysql"
```
